test/unit/run.test.js

```diff
@@ -70,4 +71,58 @@ describe('run module', function () {
             }
         ], done);
     });
+
+    it('must correctly resolve conflicts between iterationData.length and iterationCount', function (done) {
```

Comment:
By CLI level tests, I meant these: https://github.com/postmanlabs/newman/blob/develop/test/cli/suppress-exit-code.test.js

You should add a spec like that, with iteration data as a CSV file, and then run it in these combinations:
1. `--iteration-data` absent and `--iteration-count` absent (should run one iteration)
2. `--iteration-data` present and `--iteration-count` absent (should run iterations equal to number of rows in iteration data)
3. `--iteration-data` absent and `--iteration-count` present (should run iterations equal to iteration count given)
4. `--iteration-data` present and `--iteration-count` present (should always run iterations equal to given iteration count)